homeassistant/components/squeezebox/__init__.py

```diff
@@ -66,25 +82,51 @@ async def async_setup_entry(hass: HomeAssistant, entry: SqueezeboxConfigEntry) -
     _LOGGER.debug("LMS Status for setup  = %s", status)
 
     lms.uuid = status[STATUS_QUERY_UUID]
+    _LOGGER.debug("LMS %s = '%s' with uuid = %s ", lms.name, host, lms.uuid)
     lms.name = (
         (STATUS_QUERY_LIBRARYNAME in status and status[STATUS_QUERY_LIBRARYNAME])
         and status[STATUS_QUERY_LIBRARYNAME]
         or host
     )
-    _LOGGER.debug("LMS %s = '%s' with uuid = %s ", lms.name, host, lms.uuid)
+    version = STATUS_QUERY_VERSION in status and status[STATUS_QUERY_VERSION] or None
+    # mac can be missing
+    mac_connect = (
+        STATUS_QUERY_MAC in status
+        and {(CONNECTION_NETWORK_MAC, format_mac(status[STATUS_QUERY_MAC]))}
+        or None
+    )
+
+    device_registry = dr.async_get(hass)
+    device = device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        identifiers={(DOMAIN, lms.uuid)},
+        name=lms.name,
+        manufacturer=MANUFACTURER,
+        model=SERVER_MODEL,
+        sw_version=version,
```

Comment:
This shows as Firmware version the Lyrion server app doesn't have firmware. If you add the following it will make the Lyrion application a service rather than device which will then just show version.
`entry_type=DeviceEntryType.SERVICE,`